bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -507,6 +507,20 @@ protected State getScaled(Optional<? extends Number> value, Optional<Short> scal
         return getScaled(value.get().longValue(), scaleFactor.get(), unit);
     }
 
+    /**
+     * Returns value multiplied by the 10 on the power of scaleFactory
+     *
+     * @param value the value to alter
+     * @param scaleFactor the scale factor to use (may be negative)
+     * @return the scaled value as a DecimalType
+     */
+    protected State getScaled(Optional<? extends Number> value, Short scaleFactor, Unit<?> unit) {
```

Comment:
Any reason `scaleFactor` couldn't be primitive?

Reply:
In some cases the scaleFactor is also optional per spec. I didn't want to mix primitives and Optional<...>-s.
Also note, that optional values have to be treated differently if omitted: in case of a missing optional value we output the UnDefType. Any other case will output a reasonable default value.